drivers/include/si7021.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2016 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    driver_si7021 SI7021
+ * @ingroup     drivers
+ * @brief       Driver for the SI7021 temperature and humidity sensor.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition of the SI7021 driver.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ */
+
+#ifndef SI7021_H_
+#define SI7021_H_
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name SI7021 address
+ */
+#define SI7021_ADDRESS              (0x80)
+
+/**
+ * @name SI7021 device commands
+ * @{
+ */
+#define SI7021_MEASURE_RH_HOLD      (0xE5)
+#define SI7021_MEASURE_RH           (0xF5)
+#define SI7021_MEASURE_TEMP_HOLD    (0xE3)
+#define SI7021_MEASURE_TEMP         (0xF3)
+#define SI7021_MEASURE_TEMP_PREV    (0xE0)
+#define SI7021_RESET                (0xFE)
+#define SI7021_WRITE_USER_REG       (0xE6)
+#define SI7021_READ_USER_REG        (0xE7)
+#define SI7021_WRITE_HEATER_REG     (0x51)
+#define SI7021_READ_HEATER_REG      (0x11)
+#define SI7021_READ_ID_FIRST_A      (0xFA)
+#define SI7021_READ_ID_FIRST_B      (0x0F)
+#define SI7021_READ_ID_SECOND_A     (0xFC)
+#define SI7021_READ_ID_SECOND_B     (0xC9)
+#define SI7021_READ_REVISION_A      (0x84)
+#define SI7021_READ_REVISION_B      (0xB8)
+/** @} */
+
+/**
+ * @name SI7021 register values
+ * @{
+ */
+#define SI7021_ID                   (0x15)
+
+#define SI7021_REVISION_1           (0xFF)
+#define SI7021_REVISION_2           (0x20)
+/** @} */
```

Comment:
You could move the command and the register value defintion to the .c or an internal header file, as they are only used internally and not part of the drivers API...